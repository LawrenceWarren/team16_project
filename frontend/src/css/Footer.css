/*Footer main Div, everything centres correctly*/
.footerMain {
  text-align: center;
}

/*The pink banner across the top of the footer*/
.banner {
  background-color: rgb(216, 109, 127);
  margin-top: 4vw;
  margin-bottom: 0.5vw;
  color: white;
  font-size: 2.5vw;
  font-family: "Microsoft YaHei", 微软雅黑, "MicrosoftJhengHei", 华文细黑,
    STHeiti, MingLiu;
  text-align: center;
}

/*The payPal donation button*/
.donate {
  border: 0px; /*No border of any kind*/
  border-radius: 8px;
  margin-top: 0.5vw;
  margin-bottom: 0.5vw;
  height: 8.93vw; /*vw is set so that the images aspect ratio is maintained - 30:8.93*/
  width: 30vw;

  background-image: url("../resource/footer/donate.png");
  background-size: 105% 110%;
  background-repeat: no-repeat;
  background-position: center;
}

/*The div that contains the buttons*/
.SocialButtons {
  text-align: center;
  margin-top: 0.5vw;
  margin-bottom: 0.5vw;
}
/*Default button styling*/
.butt {
  margin-left: 5px;
  margin-right: 5px;
  box-shadow: -2px 2px #000000;
  border-radius: 13%;
  background-color: #ffffff;
  background-size: 100% 100%;
  background-repeat: no-repeat;
  background-position: center;
  height: 75px;
  width: 75px;
  transition: 0.3s;
}
/*Hovering over a button*/
.butt:hover {
  background-color: rgb(216, 109, 127);
  box-shadow: -2px 2px rgb(166, 59, 77);
}
/*Buttons for specific social media*/
.fb-butt {
  background-size: 95% 95%;
  background-image: url("../resource/footer/facebook.png");
}
.tw-butt {
  background-image: url("../resource/footer/ttww.png");
}
.ins-butt {
  background-size: 90% 90%;
  background-image: url("../resource/footer/ins.png");
}
.li-butt {
  background-size: 90% 90%;
  background-image: url("../resource/footer/linkedin.png");
}
.feed-butt {
  background-size: 90% 100%;
  background-image: url("../resource/footer/feedback.png");
}

/*Further info*/
.furtherInfo {
  text-align: center;
  margin-top: 2.5vw;
  margin-bottom: 0.5vw;
  font-size: 1vw;
  color: rgb(106, 103, 103);
  font-family: "Microsoft YaHei", 微软雅黑, "MicrosoftJhengHei", 华文细黑,
    STHeiti, MingLiu;
}
/*Links for further info, and states related to when they are clicked and hovered*/
.furtherInfo a {
  margin-right: 1.5vw;
  margin-left: 1.5vw;
}
.furtherInfo a:visited {
  color: rgb(106, 103, 103);
}
.furtherInfo a:hover {
  color: rgb(216, 109, 127);
}

/*CopyRight text at the bottom of the footer*/
.copyRight {
  text-align: center;
  color: rgb(106, 103, 103);
  margin-top: 0.2vw;
  font-size: 0.9vw;
}
