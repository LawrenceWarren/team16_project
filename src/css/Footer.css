/*Footer main Div, everything centres correctly*/
.footerMain {
  text-align: center;
}

/*The pink banner across the top of the footer*/
.banner {
  background-color: #c0087f;
  margin-top: 4vw;
  margin-bottom: 0.5vw;
  color: white;
  font-size: 2.5vw;
  font-family: "Microsoft YaHei", 微软雅黑, "MicrosoftJhengHei", 华文细黑,
    STHeiti, MingLiu;
  text-align: center;
}

/*The payPal donation button*/
.donate {
  border: 0px; /*No border of any kind*/
  border-radius: 8px;
  margin-top: 0.5vw;
  margin-bottom: 0.5vw;
  height: 8.93vw; /*vw is set so that the images aspect ratio is maintained - 30:8.93*/
  width: 30vw;

  background-image: url("../resource/footer/donate.png");
  background-size: 105% 110%;
  background-repeat: no-repeat;
  background-position: center;
}

/*Styling for the get involved button, and hovering over said button*/
.getInvolvedButton {
  font-size: 1.1vw;
  font-weight: bold;
  font-family: Georgia;
  height: 10vw;
  width: 30vw;
  color: #ffffff;
  background-color: #c0087f;
  margin-top: 0.5vw;
  margin-bottom: 0.5vw;
  border-radius: 5px;
  box-shadow: -2px 2px #e480ce;
}
.getInvolvedButton:hover {
  background-color: #e480ce;
  box-shadow: -2px 2px #c0087f;
}

/*The div that contains the buttons*/
.SocialButtons {
  text-align: center;
  margin-top: 0.5vw;
  margin-bottom: 0.5vw;
}
/*Default button styling*/
.butt {
  margin-left: 5px;
  margin-right: 5px;
  box-shadow: -2px 2px #000000;
  border-radius: 13%;
  background-color: #ffffff;
  background-size: 100% 100%;
  background-repeat: no-repeat;
  background-position: center;
  min-width: 10px;
  min-height: 10px;
  height: 5vw;
  width: 5vw;
  max-height: 75px;
  max-width: 75px;
  transition: 0.3s;
}
/*Hovering over a button*/
.butt:hover {
  background-color: #c0087f;
  box-shadow: -2px 2px #c0087f;
}
/*Buttons for specific social media*/
.fb-butt {
  background-size: 95% 95%;
  background-image: url("../resource/footer/facebook.png");
}
.tw-butt {
  background-image: url("../resource/footer/ttww.png");
}
.ins-butt {
  background-size: 90% 90%;
  background-image: url("../resource/footer/ins.png");
}
.li-butt {
  background-size: 90% 90%;
  background-image: url("../resource/footer/linkedin.png");
}
.feed-butt {
  background-size: 90% 100%;
  background-image: url("../resource/footer/feedback.png");
}

/*Further info*/
.furtherInfo {
  text-align: center;
  margin-top: 2.5vw;
  margin-bottom: 0.5vw;
  font-size: 1vw;
  color: rgb(106, 103, 103);
  font-family: "Microsoft YaHei", 微软雅黑, "MicrosoftJhengHei", 华文细黑,
    STHeiti, MingLiu;
}
/*Links for further info, and states related to when they are clicked and hovered*/
.furtherInfo a {
  margin-right: 1.5vw;
  margin-left: 1.5vw;
}
.furtherInfo a:visited {
  color: rgb(106, 103, 103);
}
.furtherInfo a:hover {
  color: #c0087f;
}

/*CopyRight text at the bottom of the footer*/
.copyRight {
  text-align: center;
  color: rgb(106, 103, 103);
  margin-top: 0.2vw;
  font-size: 0.9vw;
}
