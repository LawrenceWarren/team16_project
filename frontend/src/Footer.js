import React from "react";
import "./css/Footer.css";
import donate from "./resource/donate.png";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import { grey } from "@material-ui/core/colors";

const GetInvolvedButton = withStyles({
  root: {
    fontSize: "1.1vw",
    fontWeight: "bold",
    fontFamily: "Georgia",
    height: "10vw",
    width: "30vw",
    color: "rgba(39, 39, 39,0.7)",
    backgroundColor: "#FFFFFF",
    "&:hover": {
      backgroundColor: grey[400],
    },
    marginTop: "3.5vw",
  },
})(Button);

function Footer() {
  return (
    <div>
      <p className="involved"> Get Involved: </p>

      <div align="center">
        <GetInvolvedButton
          variant="contained"
          color="primary"
          onClick={() => {
            window.open("https://www.avas-angels.com/getinvolved.html");
          }}
        >
          {"Become Part Of Ava's Angels"}
        </GetInvolvedButton>
      </div>

      <div className="donate" align="center">
        <a href="https://www.paypal.com/fundraiser/112574636177901026/charity/3575409">
          <img src={donate} alt="donate with paypal or card" />
        </a>
      </div>

      <div className="SocialButtons">
        <button
          className="butt fb-butt"
          onClick={() => {
            window.open("https://www.facebook.com/avas.angels.739");
          }}
        />

        <button
          className="butt tw-butt"
          onClick={() => {
            window.open("https://twitter.com/AvasAngels_com");
          }}
        />

        <button
          className="butt ins-butt"
          onClick={() => {
            window.open("https://www.instagram.com/avasangelscharity/");
          }}
        />

        <button
          className="butt li-butt"
          onClick={() => {
            window.open("https://www.linkedin.com/in/avas-angels-1519a2160/");
          }}
        />

        <button
          className="butt feed-butt"
          onClick={() => {
            window.open("https://www.avas-angels.com/contact.html");
          }}
        />
      </div>

      <div className="related">
        <p>
          {/* eslint-disable-next-line */}
          <a href=""> FAQ </a> |{/* eslint-disable-next-line */}
          <a href=""> Terms of Use </a> |{/* eslint-disable-next-line */}
          <a href=""> Privacy </a> |{/* eslint-disable-next-line */}
          <a href=""> Contact </a>
        </p>
      </div>

      <div className="copyRight">
        <p>
          Â© Copyright 2019-2020. Designed and powered by Ali S., Lawrence W.,
          Shan W.Z., Benjamin S., Chen Y.T. and Zhang J.H. All Rights Reserved.{" "}
        </p>
      </div>
    </div>
  );
}

export default Footer;
