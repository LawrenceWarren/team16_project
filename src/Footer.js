import React from 'react';
import './css/Footer.css';
import FaceBook from "./resource/facebook.png";
import Twitter from "./resource/ttww.png";
import Ins from "./resource/ins.png";
import LinkedIn from "./resource/linkedin.png";
import FeedBack from "./resource/feedback.png";
import donate from "./resource/donate.png";
import { makeStyles, withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import {grey} from "@material-ui/core/colors";


const FooterButton = withStyles({
    root:{
        marginRight: "1.5vw",
        marginTop: "1vw",
        height: "2.5vw",
        width: "2.5vw",
        backgroundColor: "#FFFFFF",
        "&:hover": {
            backgroundColor: grey[500]
        },
    }
})(Button);

const GetInvolvedButton = withStyles({
    root:{
        fontSize: "1.1vw",
        fontWeight:"bold",
        fontFamily: "Georgia",
        height:"10vw",
        width:"30vw",
        color: "rgba(39, 39, 39,0.7)",
        backgroundColor: "#FFFFFF",
        "&:hover": {
            backgroundColor: grey[400]
        },
        marginTop: "3.5vw",
    }
})(Button);

const FaceBookImage = makeStyles(theme => ({
    image: {
        backgroundImage: `url(${FaceBook})`,
        backgroundRepeat: "no-repeat",
        backgroundSize:"50% 50%",
        backgroundPosition: "center",
    },
}));

const TwitterImage = makeStyles(theme => ({
    image: {
        backgroundImage: `url(${Twitter})`,
        backgroundRepeat: "no-repeat",
        backgroundSize:"50% 50%",
        backgroundPosition: "center",
    },
}));

const InsImage = makeStyles(theme => ({
    image: {
        backgroundImage: `url(${Ins})`,
        backgroundRepeat: "no-repeat",
        backgroundSize:"50% 50%",
        backgroundPosition: "center",
    },
}));

const LinkedInImage = makeStyles(theme => ({
    image: {
        backgroundImage: `url(${LinkedIn})`,
        backgroundRepeat: "no-repeat",
        backgroundSize:"50% 50%",
        backgroundPosition: "center",
    },
}));

const FeedBackImage = makeStyles(theme => ({
    image: {
        backgroundImage: `url(${FeedBack})`,
        backgroundRepeat: "no-repeat",
        backgroundSize:"50% 50%",
        backgroundPosition: "center",
    },
}));

function Footer(){
    const FaceBookImg = FaceBookImage();
    const TwitterImg = TwitterImage();
    const InsImg = InsImage();
    const LinkedInImg = LinkedInImage();
    const FeedBackImg = FeedBackImage();

    return(
        <div>
            <div class = "involved"> Get Involved: </div>

            <div align="center"> 
                <GetInvolvedButton 
                    variant="contained" 
                    color="primary" 
                    onClick={()=>{window.open("https://www.avas-angels.com/getinvolved.html")}}
                >
                    {"Become Part Of Ava's Angels"}
                </GetInvolvedButton>
            </div>

            <div class="donate" align="center">
                <a href="https://www.paypal.com/fundraiser/112574636177901026/charity/3575409">
                    <img src={donate} alt="donate with paypal or card"  />
                </a>
            </div>

            <div class = "contact">  
                <FooterButton 
                    variant="contained" 
                    color="primary" 
                    className={FaceBookImg.image} 
                    onClick={()=>{window.open("https://www.facebook.com/avas.angels.739")}}
                >
                </FooterButton>
                <FooterButton 
                    variant="contained" 
                    color="primary" 
                    className={TwitterImg.image} 
                    onClick={()=>{window.open("https://twitter.com/AvasAngels_com")}}
                >
                </FooterButton>
                <FooterButton 
                    variant="contained" 
                    color="primary" 
                    className={InsImg.image} 
                    onClick={()=>{window.open("https://www.instagram.com/avasangelscharity/")}}
                >
                </FooterButton>
                <FooterButton 
                    variant="contained" 
                    color="primary" 
                    className={LinkedInImg.image} 
                    onClick={()=>{window.open("https://www.linkedin.com/in/avas-angels-1519a2160/")}}
                >
                </FooterButton>
                <FooterButton 
                    variant="contained" 
                    color="primary" 
                    className={FeedBackImg.image} 
                    onClick={()=>{window.open("https://www.avas-angels.com/contact.html")}}
                >
                </FooterButton>
            </div>

            <div class = "related">
                <p>
                    {/* eslint-disable-next-line */}
                    <a href = ""> FAQ </a> |
                    {/* eslint-disable-next-line */}
                    <a href = ""> Terms of Use </a> |
                    {/* eslint-disable-next-line */}
                    <a href = ""> Privacy </a> |
                    {/* eslint-disable-next-line */}
                    <a href = ""> Contact </a>
                </p>
            </div>

            <div class = "copyRight">
                <p>Â© Copyright 2019-2020. Designed and powered by Ali S., Lawrence W., Shan W.Z., Benjamin S., Chen Y.T. and Zhang J.H. All Rights Reserved. </p>
            </div>
        </div>
    );
}

export default Footer;