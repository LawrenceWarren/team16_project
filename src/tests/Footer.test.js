//These tests were written by Wenzheng Shan.

import React from "react";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import Footer from "../Footer";

enzyme.configure({ adapter: new Adapter() });

//  SUITE 1: Snapshot rendering tests
describe("Footer Component status: ", () => {
  //  TEST 1: If the footer component has been updated since last snapshot
  it("If the <Footer /> component has been updated.", () => {
    //Create a snap shot of footer page
    const tree = renderer.create(<Footer />).toJSON();

    //Compares it to an old snapshot
    expect(tree).toMatchSnapshot();
  });
});

//  SUITE 2: HEADER TEST
describe("Header Test: ", () => {
  //  TEST 1: If the footer head is Get Involved
  it("If the footer head is Get Involved.", () => {
    const footer = enzyme.shallow(<Footer />);
    const head = footer.find(".banner");
    expect(head.text()).toBe(" Get Involved: ");
  });
});

//  SUITE 3: BUTTON NUMBER TEST
describe("Button Number Test: ", () => {
  const footer = enzyme.shallow(<Footer />);
  //  TEST 1: The number of donate Button should be 1
  it("The number of donate button.", () => {
    const button = footer.find("button");
    expect(button.find(".donate").length).toBe(1);
  });

  //  TEST 2: The number of social media button should be 5
  it("The number of social media button.", () => {
    const SocialButton = footer.find(".SocialButtons");
    expect(SocialButton.children().length).toBe(5);
  });

  //  TEST 3: The number of further information button should be 4
  it("The number of further information button", () => {
    expect(footer.find("a").length).toBe(4);
  });
});
