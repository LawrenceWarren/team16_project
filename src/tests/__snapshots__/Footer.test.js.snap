// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component status:  If the <Footer /> component has been updated. 1`] = `
<div
  className="footerMain"
>
  <p
    className="banner"
  >
     Get Involved: 
  </p>
  <div
    align="center"
  >
    <button
      className="getInvolvedButton"
      onClick={[Function]}
    >
      Become Part Of Ava's Angels
    </button>
  </div>
  <button
    className="donate"
    onClick={[Function]}
  />
  <div
    className="SocialButtons"
  >
    <button
      className="butt fb-butt"
      onClick={[Function]}
    />
    <button
      className="butt tw-butt"
      onClick={[Function]}
    />
    <button
      className="butt ins-butt"
      onClick={[Function]}
    />
    <button
      className="butt li-butt"
      onClick={[Function]}
    />
    <button
      className="butt feed-butt"
      onClick={[Function]}
    />
  </div>
  <div
    className="furtherInfo"
  >
    | 
    <a
      href="https://www.google.com"
    >
       FAQ 
    </a>
     | 
    <a
      href="https://www.google.com"
    >
       Terms of Use 
    </a>
     |
    <a
      href="https://www.google.com"
    >
       Privacy 
    </a>
     |
    <a
      href="https://www.google.com"
    >
       Contact 
    </a>
     |
  </div>
  <div
    className="copyRight"
  >
    <p>
      Â© Copyright 2019-2020. Designed and powered by Ali S., Lawrence W., Shan W.Z., Benjamin S., Chen Y.T. and Zhang J.H. All Rights Reserved.
       
    </p>
  </div>
</div>
`;
